cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompilerImpl.java

```diff
@@ -65,10 +65,28 @@ public Executable preCompile(SlangSource source) {
         // Then we transform the parsed Slang source to a Slang model
         Executable executable = slangModeller.createModel(parsedSlang);
         Validate.notEmpty(executable.getNamespace(),"Operation/Flow " +executable.getName() +" must have a namespace");
+        ValidateFileName(source, parsedSlang);
 
         return executable;
     }
 
+    private void ValidateFileName(SlangSource source, ParsedSlang parsedSlang) {
+        String fileName = source.getName().substring(0, source.getName().lastIndexOf("."));
+        String flowOrOperationName = getFlowOrOperationName(parsedSlang);
+        Validate.isTrue(fileName.equals(flowOrOperationName),
```

Comment:
you should move the validation logic in SlangModellerImpl.transformToExecutable() where you already have the necessary data: ParsedSlang.getnName() and exeutableName